exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -197,6 +196,223 @@ func TestGaugeConsumerMissingValue(t *testing.T) {
 	assert.Len(t, allLogs, expectedMissingValueCount)
 }
 
+func TestSumConsumerDelta(t *testing.T) {
+	deltaMetric := newMetric(
+		"test.delta.metric", pdata.MetricDataTypeSum)
+	sum := deltaMetric.Sum()
+	sum.SetAggregationTemporality(pdata.MetricAggregationTemporalityDelta)
+	dataPoints := sum.DataPoints()
+	dataPoints.EnsureCapacity(2)
+	addDataPoint(
+		35,
+		1635205001,
+		map[string]interface{}{
+			"env": "dev",
+		},
+		dataPoints,
+	)
+	addDataPoint(
+		52.375,
+		1635205002,
+		map[string]interface{}{
+			"env": "prod",
+		},
+		dataPoints,
+	)
+
+	sender := &mockSumSender{}
+	consumer := newSumConsumer(sender, nil)
+	assert.Equal(t, pdata.MetricDataTypeSum, consumer.Type())
+	var errs []error
+
+	// delta sums get treated as delta counters
+	consumer.Consume(deltaMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	expected := []tobsMetric{
+		{
+			Name:  "test.delta.metric",
+			Value: 35.0,
+			Tags:  map[string]string{"env": "dev"},
+		},
+		{
+			Name:  "test.delta.metric",
+			Value: 52.375,
+			Tags:  map[string]string{"env": "prod"},
+		},
+	}
+	assert.ElementsMatch(t, expected, sender.deltaMetrics)
```

Comment:
Can we add an assertion like `assert.Empty(t, sender.metrics)`? And add corresponding assertions in the other tests so that both `sender.deltaMetrics` and `sender.metrics` are checked.

Reply:
Done.